src/profile-logic/import/dhat.js

```diff
@@ -267,20 +275,16 @@ export function attemptToConvertDhat(json: mixed): Profile | null {
         'Expected to find a funcIndex from a frameIndex'
       );
 
-      // Case 1: The stack index starts at -1, increment by 1 to start searching stacks
-      //         at index 0.
-      // Case 2: This is the previously matched stack index, increment it by 1 to continue
-      //         searching at the next stack index.
-      stackIndex++;
-
-      // Start searching for a stack index.
-      for (; stackIndex < stackTable.length; stackIndex++) {
-        const nextFrameIndex = stackTable.frame[stackIndex];
-        if (
-          frameTable.func[nextFrameIndex] === funcIndex &&
-          stackTable.prefix[stackIndex] === prefix
-        ) {
-          break;
+      if (candidateStackTables) {
+        // Start searching for a stack index.
+        stackIndex = stackTable.length;
```

Comment:
nit: I think I'd convert this to `let stackIndex = null` right before the `if`. Then in the `if` below we can check for `if (stackIndex === null) { stackIndex = stackTable.length; ... }`.  I believe this would make this code more explicit. What do you think?

Reply:
While implementing your other suggestion (https://github.com/firefox-devtools/profiler/pull/4640#discussion_r1210265862), I found out that it will only work if we use `stackIndex = stackTable.length` every time, not just when entering this loop.

This is due to the old value of `stackIndex`, from a previous iteration of the loop, not being equal to `stackTable.length`, while `candidateStackTables` is `undefined` (so, we are in this prefix for the first time). So, it will neither enter this loop, nor create a new stackTable entry, which causes issues.

Because of that, I think it's best to simply move this `stackIndex = stackTable.length` line up, and add a comment explaining that that is a fallback when we don't enter this loop, or we enter it but don't find a match.

Reply:
Mmm I feel like my suggestion here would still work but what you did works too, and the comment makes it clear enough, so I don't mind.